### DISCUSSION TITLE
Find component by id

### DISCUSSION THREAD

**Discussion by ArtMin96** [livewire/livewire#5483] [1. Help]
How to find the component by id (not js)?

**Comment by RealMrHex** [comment]
What do you mean of "**NOT JS**"?
Do you want to find component id from your **Livewire** component using **PHP**?
Why do you need something like this?
I think there are better ways to do it if you can explain your case

**Reply by ArtMin96** [reply]
I have a `Posts` component and a `SinglePost` component, I want to update only that `SinglePost` component when I perform an action in another component that is also inside the `SinglePost` component. But I have already done it.

```php
class SinglePost extends Component
{
    public Post $post;

    protected function getListeners(): array
    {
        return [
            "refresh-single-post-{$this->post->id}" => '$refresh',
        ];
    }

    //...
}
```

```php
class React extends Component
{
    public Reacterable $reacter;

    public Reactable $reactant;

    public function react()
    {
        //...

        $this->emit('refresh-single-post-'.$this->reactant->id);
    }

    //...
}
```

**Reply by ArtMin96** [reply]
I think in js you can use 
```js
window.livewire.find('component-id');

// Or

component.get('component-id');
```

**Reply by RealMrHex** [reply]
So there is no need to find component by id if I understand what you mean!
Simply fire some event using ```emitTo()``` to the single component and perform any action you want on the event listener.
Also you can pass some external attributes such as post-id and use it in the listener for detecting the component that should be updated.
feel free to update me if I miss-understood your case.
